src/control-plane/backend/lambda/api/service/stack.ts

```diff
@@ -322,6 +326,48 @@ export class StackManager {
     return state;
   }
 
+  public async resetIngestionStackTemplate(templateUrl: string, templateVersion: string) {
+    if (!this.execWorkflow || !this.workflow) {
+      throw new Error('Pipeline workflow is empty.');
+    }
+    this.execWorkflow.Workflow = this._resetIngestionStackTemplate(this.execWorkflow.Workflow, templateUrl, templateVersion);
+    this.workflow.Workflow = this._resetIngestionStackTemplate(this.workflow.Workflow, templateUrl, templateVersion);
+  }
+
+  private _resetIngestionStackTemplate(state: WorkflowState, templateUrl: string, templateVersion: string): WorkflowState {
+    if (state.Type === WorkflowStateType.PARALLEL) {
+      for (let branch of state.Branches as WorkflowParallelBranch[]) {
+        for (let key of Object.keys(branch.States)) {
+          branch.States[key] = this._resetIngestionStackTemplate(branch.States[key], templateUrl, templateVersion);
+        }
+      }
+    } else if (state.Type === WorkflowStateType.STACK && state.Data?.Input.StackName) {
+      if (state.Data.Input.StackName ===
+        getStackName(this.pipeline.pipelineId, PipelineStackType.INGESTION, this.pipeline.ingestionServer.sinkType)) {
+        state.Data.Input.TemplateURL = templateUrl;
+        state.Data.Input.Parameters = removeParameters(
```

Comment:
why we can not use the annotation on fields?

Reply:
There are too many logic changed, if using annotation.